casper/src/main/resources/PoS.rhox

```diff
@@ -155,26 +154,64 @@ new  PoS,
               }
             } |
 
-            contract PoS (@"chargeDeploy", @deployerId, @amount, returnCh) = {
-              new vaultCh, transferCh, userCh, processPayCh,
-                  depositCh,
-                  newPendingCh in {
-                getUser!(deployerId, *userCh) |
-                runMVar!(*stateCh, *processPayCh, *returnCh) |
-                for(@state, payResultCh <- processPayCh;
-                    @userPk <- userCh) {
+            new currentDeployerData in {
+              contract PoS (@"chargeDeploy", @deployerId, @amount, return) = {
+                new depositCh in {
                   deposit!(deployerId, amount, *depositCh) |
-
-                  distributeRewards!(
-                    amount,
-                    state.get("activeValidators"),
-                    state.get("allBonds"),
-                    state.get("pendingRewards"),
-                    *newPendingCh) |
-
-                  for(@depositResult <- depositCh;
-                      @newPending <- newPendingCh) {
-                    payResultCh!(state.set("pendingRewards", newPending), depositResult)
+                  for(@depositResult <- depositCh){
+                    match depositResult {
+                      (true, _) => {
+                        currentDeployerData!((deployerId, amount)) |
```

Comment:
Since the `currentDeployerData` channel would be shared by all users, we won't be able to process deploys concurrently.
Are we not able to do that anyway or what are your thoughts here?

Reply:
So you are complaining about a mergeability issue. I think this has a lower priority at this point but we should probably add a Jira item to address this issue

Reply:
Mergeability is about blocks getting merged. My question here is more about concurrent deploy processing in one block.

Reply:
> Are we not able to do that anyway or what are your thoughts here?

That's right. There are fundamental problems preventing us from processing deploys in parallel in the current architecture, so I don't really think that's an issue.

Reply:
Out of curiosity: what are the other issues?

Reply:
Say I have `n` deploys and one throws an error. Right now, we have an opportunity to revert state before the next deploy is processed. If all deploys were processed in parallel, we'd have to cancel the entire computation, revert the state, extract the erring deploy, and run the sequence again without the erring process.